src/hera/workflows/_runner/script_annotations_util.py

```diff
@@ -46,6 +46,12 @@ def _get_dumper_function(destination: Union[Parameter, Artifact]) -> Callable:
     if isinstance(destination, Parameter) and destination.dumps is not None:
         return destination.dumps
 
+    if isinstance(destination, Artifact):
+        if destination.dumps is not None:
+            return destination.dumps
+        if destination.dump is not None:
+            return destination.dump
```

Comment:
I'm not comfortable with how this is discarding whether the expected input is str or not :/ It doesn't seem like the call site can have enough information to call this safely.

Reply:
Not sure if you mixed up input/output or dumping/loading itself there - if you meant output, then I don't _think_ it matters - we just call the `dumper` function passed to `_write_to_path` and see what it returned, then call `path.write_text` or `path.write_bytes`.

Reply:
Yep, I did.